compiler/src/org.graalvm.compiler.replacements/src/org/graalvm/compiler/replacements/StandardGraphBuilderPlugins.java

```diff
@@ -2282,4 +2283,25 @@ public SnippetTemplate.SnippetInfo getSnippet(BigIntegerSnippets.Templates templ
             }
         });
     }
+
+    private static boolean hasEnsureMaterializedForStackWalk() {
+        try {
+            Class.forName("java.lang.Thread").getDeclaredMethod("ensureMaterializedForStackWalk");
+        } catch (Exception e) {
+            return false;
+        }
+        return true;
+    }
+
+    private static void registerThreadPlugin(InvocationPlugins plugins, Replacements replacements) {
+        InvocationPlugin threadPlugin = new InvocationPlugin("ensureMaterializedForStackWalk", Object.class) {
+            @Override
+            public boolean apply(GraphBuilderContext b, ResolvedJavaMethod targetMethod, Receiver receiver, ValueNode object) {
+                b.add(new BlackholeNode(object));
+                return true;
+            }
+        };
+        Registration r = new Registration(plugins, "java.lang.Thread", replacements);
+        r.registerConditional(hasEnsureMaterializedForStackWalk(), threadPlugin);
+    }
```

Comment:
@mur47x111 @dougxc this intrinsic would also be used on HotSpot, please review.

Reply:
Looks good. This is the same change @gilles-duboscq made on a similar PR.